opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -442,59 +600,47 @@ def __init__(
         record_exception: bool = True,
         set_status_on_exception: bool = True,
     ) -> None:
-
-        self.name = name
-        self.context = context
-        self.parent = parent
-        self.sampler = sampler
-        self.trace_config = trace_config
-        self.resource = resource
-        self.kind = kind
+        super().__init__(
+            name=name,
+            context=context,
+            parent=parent,
+            kind=kind,
+            resource=resource,
+            instrumentation_info=instrumentation_info,
+        )
+        self._sampler = sampler
+        self._trace_config = trace_config
         self._record_exception = record_exception
         self._set_status_on_exception = set_status_on_exception
-
-        self.span_processor = span_processor
-        self.status = Status(StatusCode.UNSET)
+        self._span_processor = span_processor
         self._lock = threading.Lock()
 
         _filter_attribute_values(attributes)
         if not attributes:
-            self.attributes = self._new_attributes()
+            self._attributes = self._new_attributes()
         else:
-            self.attributes = BoundedDict.from_map(
+            self._attributes = BoundedDict.from_map(
                 SPAN_ATTRIBUTE_COUNT_LIMIT, attributes
             )
 
-        self.events = self._new_events()
+        self._events = self._new_events()
         if events:
             for event in events:
                 _filter_attribute_values(event.attributes)
                 # pylint: disable=protected-access
                 event._attributes = _create_immutable_attributes(
                     event.attributes
                 )
-                self.events.append(event)
+                self._events.append(event)
 
         if links is None:
-            self.links = self._new_links()
+            self._links = self._new_links()
```

Comment:
Since `links` defaults to `()`, I don't think this instantiation of a bounded list will ever run. We should just default links to `None` like how events works.

Reply:
Updated